snapshotter/pkg/dmsetup/dmsetup.go

```diff
@@ -258,6 +258,55 @@ func Version() (string, error) {
 	return dmsetup("version")
 }
 
+// DeviceStatus represents devmapper device status information
+type DeviceStatus struct {
+	Offset int64
+	Length int64
+	Target string
+	Params []string
+}
+
+// Status provides status information for devmapper device
+func Status(deviceName string) (*DeviceStatus, error) {
+	var (
+		err    error
+		status DeviceStatus
+	)
+
+	output, err := dmsetup("status", deviceName)
+	if err != nil {
+		return nil, err
+	}
+
+	output = strings.Trim(output, "\n")
```

Comment:
Doesn't `dmsetup` already trim `\n`? Is there multiple expected new lines?

Reply:
`dmsetup status` can output multiple lines. Here I want to make sure that the last parameter doesn't contain `\n` at the end

Reply:
I think any parsing of the `dmsetup` function should be contained there. How do you feel about moving this in `dmsetup` with an additional if statement?

```go
if action == "status" {
	output = strings.Trim(output, "\n")
}
```

Reply:
Actually `dmsetup` already [contains](https://github.com/firecracker-microvm/firecracker-containerd/blob/81542ccb06325417a21faa19dd0ca87ecd4d9b31/snapshotter/pkg/dmsetup/dmsetup.go#L343) `strings.TrimSuffix(output, "\n")`. It makes sense to replace it with `strings.Trim(output, "\n")`

Reply:
@mxpv - Yea, that's why I was confused :D

Reply:
Done